### PR TITLE
Ensure toggle anchors work with the spacebar and are not draggable

### PR DESCRIPTION
**Pull request by nickcolley**
This pull request introduces two accessibility improvements:

## Making sure anchor element toggle behaves the same as button element when pressing space.
This is because some assistive technologies, such as screen readers, will announce that you can activate a button control by pressing the space bar.
This works fine with a native button as you can activate it with both enter and space, but requires additional JavaScript for links with role=button.

## Making sure anchor elements are not draggable to make them more reliable
If a user drags their mouse slightly when trying to interact with the toggle it will trigger the 'drag and drop' functionality.
Regular buttons prevent this and ensure as long as the mouse is still above the button that the click will register.
Some people have motor impairments which result in a tremor, this means they can have involuntary movements or shaking that makes using a mouse or trackpad difficult.

This is based on [research done by the GOV.UK Notify team](https://github.com/alphagov/govuk-frontend/pull/1020#issue-219981238).

When investigating https://github.com/Financial-Times/o-header/pull/384 I wanted to do some more work to ensure the menu anchors are not draggable but then found that this code is defined in `o-toggle` not `o-header`. Sorry for the mixup...

I think there is a risk if we release this that it'll break the header as we're doing it twice, so if you choose to accept this we need to consider the order we release. Perhaps reverting the [header specific functionality first](https://github.com/Financial-Times/o-header/pull/384).

### CODE REVIEW

**Comment by nickcolley** [comment]
I need to fix linting errors, just found out the command...

File: origami.json [hunk 1]
```diff
@@ -41,6 +41,13 @@
         "type": "visibility"
       },
       "description": "This toggle target has a class \"o-toggle-visibility\" which visually hides it, whilst still occupying space, until clicking the toggle updates its aria attributes and adds the \"o-toggle--active\" class."
-    }
+    },
+		{
+			"hidden": true,
+			"title": "Anchor toggle",
+			"name": "anchor",
+      "template": "demos/src/anchor.mustache",
```
Comment (JakeChampion) [inline origami.json:49]:
```suggestion
			"template": "demos/src/anchor.mustache",
```

**Comment by JakeChampion** [comment]
Thanks you so much for this @nickcolley  <3

**Comment by nickcolley** [comment]
@JakeChampion what do you think about the roll out?

> I think there is a risk if we release this that it'll break the header as we're doing it twice, so if you choose to accept this we need to consider the other we release. Perhaps reverting the header specific functionality first.

I'm not sure how bower dependencies resolve, if we released this as a patch version would it automatically get picked up by `o-header` potentially breaking it?

**Comment by JakeChampion** [comment]
yes, we would need to revert the o-header functionality before releasing this work :-)

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v2.1.0 :tada:

**Pull request merged**